README.md

```diff
@@ -1,8 +1,95 @@
 # k3a-lag-exporter
 
-**WARNING!**
+A very simple lag exporter for Kafka topic partitions, where "lag" is
+defined as the difference between the last produced index and the last
+consumed index.
 
-This is experimental work in very early stages.
+This implementation is intended to be run in a K8s-like environment,
+where pods are restarted on failure. Hence, any exception during
+runtime will make the application terminate.
 
-* Almost no tests
-* Not at all robust, will terminate from the slightest failure
+The lag will be exported as a metric for Prometheus, like this:
+
+```text
+k3a_consumergroup_group_lag{cluster_name="the-cluster", group="the-group", partition="0", topic="the-topic"} 0
+```
+
+There's also a metric reporting the time spent polling the cluster:
+
+```text
+k3a_lag_exporter_poll_time_ms{cluster_name="the-cluster"} 8
+```
+
+Example configuration:
+
+```text
+k3a-lag-exporter {
+    poll-interval = 30 seconds
+    reporters.prometheus.port = 8000
+    clusters = [
+    {
+        name = "the-kafka-cluster"
+        topic-allow-list = [
+            "topic1"
+            "topic2"
+        ]
+        topic-deny-list = [
+            ".*secret-topic.*"
+        ]
+        group-allow-list = [
+            "public-group.*"
+        ]
+        group-deny-list = [
+            "internal-group.*"
+        ]
+        bootstrap-servers = "kafka.example.com:9092"
+        consumer-properties = {
+            security.protocol = "SASL_SSL"
+            sasl.mechanism = "PLAIN"
+            sasl.jaas.config = "org.apache.kafka.common.security.plain.PlainLoginModule required username='"${USER}"' password='"${PASSWORD}"';"
+        }
+        admin-properties = {
+            security.protocol = SSL
+            ssl.keystore.type = "PKCS12"
+            ssl.keystore.location = "./foo.jks"
+            ssl.keystore.password = "password"
+        }
+    }
+    ]
+}
+```
+
+Configuration is handled by
+[lightbend/config](https://github.com/lightbend/config).
```

Comment:
Is this correct?

Reply:
Yes. But the maven artifact is `com.typesafe`.